gensim/test/test_poincare.py

```diff
@@ -0,0 +1,173 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Author: Jayant Jain <jayantjain1992@gmail.com>
+# Copyright (C) 2017 Radim Rehurek <me@radimrehurek.com>
+# Licensed under the GNU LGPL v2.1 - http://www.gnu.org/licenses/lgpl.html
+
+"""
+Automated tests for checking the poincare module from the models package.
+"""
+
+import logging
+import os
+import tempfile
+import unittest
+from unittest.mock import Mock
+
+import numpy as np
+
+from gensim.models.poincare import PoincareData, PoincareModel
+
+
+module_path = os.path.dirname(__file__)  # needed because sample data files are located in the same folder
+datapath = lambda fname: os.path.join(module_path, 'test_data', fname)
+logger = logging.getLogger(__name__)
+
+
+def testfile():
+    # temporary data will be stored to this file
+    return os.path.join(tempfile.gettempdir(), 'gensim_word2vec.tst')
+
+
+class TestPoincareData(unittest.TestCase):
+    def test_encoding_handling(self):
+        """Tests whether utf8 and non-utf8 data loaded correctly."""
+        non_utf8_file = datapath('poincare_cp852.tsv')
+        relations = [relation for relation in PoincareData(non_utf8_file, encoding='cp852')]
+        self.assertEqual(len(relations), 2)
+        self.assertEqual(relations[0], (u'tímto', u'budeš'))
+
+        utf8_file = datapath('poincare_utf8.tsv')
+        relations = [relation for relation in PoincareData(utf8_file)]
+        self.assertEqual(len(relations), 2)
+        self.assertEqual(relations[0], (u'tímto', u'budeš'))
+
+
+class TestPoincareModel(unittest.TestCase):
```

Comment:
Nice test coverage!

What I'm missing is a test that checks the embedding actually works. Would it be possible to train on some very small data, such as this graph:
```
A
| \
B   C
|\  |\
D E F G
```
and assert, for instance, that `distance(A, B) < distance(A, D)`?

Reply:
I agree, I plan on adding this, it relies on the `PoincareKeyedVector` operations though, which are in the `poincare_model_keyedvectors` branch.